src/networks/invertible_network_conditional_hint_multiscale.jl

```diff
@@ -102,12 +103,15 @@ NetworkMultiScaleConditionalHINT3D(args...;kw...) = NetworkMultiScaleConditional
 function cat_states(XY_save::AbstractArray{Array, 2}, X::AbstractArray{Float32, 4}, Y::AbstractArray{Float32, 4})
     X_full = []
     Y_full = []
+
     for j=1:size(XY_save, 1)
         X_full = cat(X_full, vec(XY_save[j, 1]); dims=1)
         Y_full = cat(Y_full, vec(XY_save[j, 2]); dims=1)
     end
-    X_full = cat(X_full, vec(X); dims=1)
+    X_full = cat(X_full, vec(X); dims=1) #NOTE THIS MAKES THINGS INTO A NORMAL non-cuda ARRAY. 
```

Comment:
At this point (in gpus), X is a CUDA array while X_full is not. then after the cat it all becomes a non-cuda array. I need to figure out how to fix this.

Reply:
`XY_save::AbstractArray{Array, 2}` this prevents gpu already so that's an issue, then `X_full = []` not CUDA neither. So you need some one line cat of all j or initialize it properly.